src/main/java/org/skriptlang/skript/elements/expressions/ExprRotateQuaternion.java

```diff
@@ -0,0 +1,153 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.expressions;
+
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+import org.joml.Quaternionf;
+import org.joml.Vector3f;
+import org.skriptlang.skript.lang.util.JomlBukkitUtils;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("Rotate Quaternion/Vector")
+@Description({
+	"Rotates a Quaternion around an axis a set amount of degrees or rotated by a vector.",
+	"When you use a vector, the values will be used to rotate the x, y and z.",
+	"Vector numbers will get converted from a double down to a float."
+})
+@Examples("set {_new} to {_quaternion} rotated by vector(1, 0, 0)")
```

Comment:
1, 0, 0 is a bit confusing, and there should be an example for the other pattern too.

```suggestion
@Examples("set {_new} to {_quaternion} rotated by vector(90, 0, 45)")
```